$NetBSD: patch-src_libstd_build.rs,v 1.3 2019/01/19 12:44:08 ryoon Exp $

SunOS support:
 - Support PKGSRC_USE_SSP (ugly for now).
 - Add libumem support.
 - Add libexecinfo for backtrace().

diff -wpruN '--exclude=*.orig' a~/src/libstd/build.rs a/src/libstd/build.rs
--- rustc-1.40.0-src/src/libstd/build.rs	1970-01-01 00:00:00
+++ rustc-1.40.0-src/src/libstd/build.rs	1970-01-01 00:00:00
@@ -16,6 +16,7 @@ fn main() {
         println!("cargo:rustc-link-lib=execinfo");
         println!("cargo:rustc-link-lib=pthread");
     } else if target.contains("netbsd") {
+        println!("cargo:rustc-link-lib=execinfo");
         println!("cargo:rustc-link-lib=pthread");
         println!("cargo:rustc-link-lib=rt");
     } else if target.contains("dragonfly") || target.contains("openbsd") {
@@ -25,6 +26,10 @@ fn main() {
         println!("cargo:rustc-link-lib=posix4");
         println!("cargo:rustc-link-lib=pthread");
         println!("cargo:rustc-link-lib=resolv");
+        println!("cargo:rustc-link-lib=nsl");
+        // pkgsrc hack until I can figure out how to pass it through properly
+        println!("cargo:rustc-link-lib=ssp");
+        println!("cargo:rustc-link-lib=umem");
     } else if target.contains("apple-darwin") {
         println!("cargo:rustc-link-lib=System");
 
